sui-execution/src/lib.rs

```diff
@@ -21,7 +22,13 @@ pub fn executor(
 ) -> SuiResult<Arc<dyn Executor + Send + Sync>> {
     let version = protocol_config.execution_version_as_option().unwrap_or(0);
     Ok(match version {
-        0 => Arc::new(latest::Executor::new(
+        0 => Arc::new(v0::Executor::new(
+            protocol_config,
+            paranoid_type_checks,
+            silent,
+        )?),
+
+        1 => Arc::new(latest::Executor::new(
```

Comment:
I suppose the idea is that latest is always going to be `previous_version + 1`?
Naively I would have thought they are all stable numbers and so `latest` would be `0` and we go from `1` on.
But maybe not an issue?

Reply:
All the other versions need to have stable numbers, but in the current scheme, latest cannot.  In the current scheme, what I have done is created `v0` which represents the snapshot of `latest` that we have been using up until now.  That means that `v0` needs to adopt the execution version in the protocol config that `latest` was using, and `latest` therefore needs a new version.

We don't have to be so prescriptive as to say that `latest = previous_version + 1`, because we might have had temporary feature branches introduced in the meanwhile (although we could get around them interfering in lots of ways: Using the `Chain` enum to distinguish what versions mean on devnet vs production networks, or counting temporary version numbers from `u64::max()` down, rather than up from `latest`), but it does strike me as the least confusing thing to do to make sure that `vN` maps to execution version `N` on testnet and mainnet.